azurelinuxagent/common/protocol/metadata.py

```diff
@@ -170,7 +170,14 @@ def detect(self):
                                 "{0}.crt".format(thumbprint))
         shutil.copyfile(trans_prv_file, prv_file)
         shutil.copyfile(trans_cert_file, crt_file)
-        self.update_goal_state()
+        self._update_certs_with_retry()
```

Comment:
The main call to update_goal_state is now in the main loop, which is protocol-independent, so I added a new method update_goal_state to the Protocol interface.

Metadata already had an update_goal_state, which just updates the certificates; I renamed that implementation to _update_certs_with_retry.